docs/api/target.rst

```diff
@@ -118,52 +118,88 @@ blt_add_library
-needed by any target that is dependent on this library.  These will
-be inherited by CMake's target dependency rules.
+DEFINES
+  List of compiler defines both used by this target and inherited by dependent
+  targets
 
-If given a DEPENDS_ON argument, it will add the necessary includes and 
-libraries if they are already registered with blt_register_library.  If 
-not it will add them as a CMake target dependency.
+DEPENDS_ON
+  List of CMake targets and BLT registered libraries that this library
+  depends on
 
-In addition, this macro will add the associated dependencies to the given
-library target. Specifically, it will add the dependency for the CMake target
-and for copying the headers for that target as well.
+OUTPUT_NAME
+  Override built file name of library (defaults to <NAME>)  
 
-The OUTPUT_DIR is used to control the build output directory of this 
-library. This is used to overwrite the default lib directory.
-OUTPUT_NAME is the name of the output file; the default is NAME.
+OUTPUT_DIR
+  Directory that this target will built to
 
-It's useful when multiple libraries with the same name need to be created
-by different targets. NAME is the target name, OUTPUT_NAME is the library name.
-CLEAR_PREFIX allows you to remove the automatically appended "lib" prefix
-from your built library.  The created library will be foo.a instead of libfoo.a.
-FOLDER is an optional keyword to organize the target into a folder in an IDE.
+SHARED
+  Builds library as shared and overrides global BUILD_SHARED_LIBS (defaults to OFF)
+
+OBJECT
+  Create an Object library
+
+CLEAR_PREFIX
+  Removes library prefix (defaults to 'lib' on linux)
+
+FOLDER
+  Name of the IDE folder to ease organization
 
-This is available when ENABLE_FOLDERS is ON and when the cmake generator
-supports this feature and will otherwise be ignored. 
+This macro supports three types of libraries automatically: normal, header-only,
+or object.
 
-Note: Do not use with header-only (INTERFACE) libraries, as this will generate 
-a CMake configuration error.
+Normal libraries are libraries that have sources that are compiled and linked into a single
+library and have headers that go along with them (unless it's a Fortran library).
+
+Header-only libraries are useful when you do not the library separately compiled or 
+are using C++ templates that require the library's user to instatiate them. To create
+a header-only library (CMake calls them INTERFACE libraries), simply
+list all headers under the HEADER argument and do not specify SOURCES.
+
+Object libraries are basically a collection of compiled source files that are not
+archived or linked. They are sometimes useful when you want to solve compilicated linking
+problems (like circular dependencies) or when you want to combine smaller libraries into
+one larger library but don't want the linker to remove unused symbols. Unlike regular CMake
+object libraries you do not have to use the $<TARGET_OBJECTS:<libname>> syntax, you can just
```

Comment:
> $<TARGET_OBJECTS:<libname>>

This might look better in the generated sphinx docs if you surround it with double ticks, e.g.

> \`\`$<TARGET_OBJECTS:<libname>>\`\`